lib/gemfile_helper.rb

```diff
@@ -19,8 +19,33 @@ def load_dotenv
                               )
     end
 
+    GEM_NAME = '[A-Za-z0-9\.\-\_]+'.freeze
+    GEM_OPTIONS = '(.+?)\s*(?:,\s*(.+?)){0,1}'.freeze
+    GEM_SEPARATOR = '\s*(?:,|\z)'.freeze
+    GEM_REGULAR_EXPRESSION = /(#{GEM_NAME})(?:\(#{GEM_OPTIONS}\)){0,1}#{GEM_SEPARATOR}/
+
+    def parse_agent_gems(string)
+      return unless string
+      string.scan(GEM_REGULAR_EXPRESSION).each do |name, version, args|
+        if version =~ /\w+:/
+          args = "#{version},#{args}"
+          version = nil
+        end
+        yield [name, version, parse_gem_args(args)].compact
+      end
+    end
+
     private
 
+    def parse_gem_args(args)
+      return nil unless args
+      options = {}
+      args.scan(/(\w+):\s*(.+?)#{GEM_SEPARATOR}/).each do |key, value|
```

Comment:
Maybe `[^,]+?` so it skips the separator, or does it not matter because of the `?`?